src/org/zaproxy/zap/extension/viewstate/HttpPanelViewStateView.java

```diff
@@ -0,0 +1,268 @@
+import org.zaproxy.zap.extension.httppanel.view.HttpPanelView;
+import org.zaproxy.zap.extension.httppanel.view.HttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.HttpPanelViewModelEvent;
+import org.zaproxy.zap.extension.httppanel.view.HttpPanelViewModelListener;
+import org.zaproxy.zap.extension.viewstate.ViewStateModel.ViewStateUpdatedListener;
+import org.zaproxy.zap.extension.viewstate.zap.utils.ASPViewState;
+import org.zaproxy.zap.extension.viewstate.zap.utils.ViewState;
+import org.zaproxy.zap.utils.DisplayUtils;
+
+public class HttpPanelViewStateView implements HttpPanelView, HttpPanelViewModelListener, ViewStateUpdatedListener {
+
+	public static final String NAME = "HttpPanelViewStateView";
+	private static final String CAPTION_NAME = Constant.messages.getString("viewstate.panel.caption");
+	private static Logger logger = Logger.getLogger(ExtensionHttpPanelViewStateView.class);
+	private HttpPanelViewStateModel httpPanelHexModel = null;
+	private JTable hexTableBody = null;
+	private JPanel container = null;
+	private JFormattedTextField vsInfoTxt = null;
+	private javax.swing.JScrollPane scrollHexTableBody = null;
+	private boolean isEditable = false;
+	private AbstractByteHttpPanelViewModel model;
+	private boolean isEnabled = true;
+	
+	public HttpPanelViewStateView(AbstractByteHttpPanelViewModel model, boolean isEditable) {
+		this.model = model;
+		
+		getHttpPanelHexModel().setEditable(isEditable);
+		
+		this.model.addHttpPanelViewModelListener(this);
+	}
+	
+	@Override
+	public String getName() {
+		return NAME;
+	}
+	
+	@Override
+	public String getCaptionName() {
+		return CAPTION_NAME;
+	}
+	
+	@Override
+	public String getTargetViewName() {
+		return "";
+	}
+	
+	@Override
+	public int getPosition() {
+		return 0;
+	}
+
+	@Override
+	public JScrollPane getPane() {
+		if (scrollHexTableBody == null) {
+			scrollHexTableBody = new javax.swing.JScrollPane();
+			scrollHexTableBody.setName(CAPTION_NAME);
+			scrollHexTableBody.setViewportView(getContainerPanel());
+			scrollHexTableBody.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_NEVER);
+		}
+		return scrollHexTableBody;
+	}
+	
+	private JPanel getContainerPanel() {
+		if (container == null) {
+			container = new JPanel();
+			container.setLayout(new BoxLayout(container, BoxLayout.PAGE_AXIS));
+			JScrollPane cScroll = new JScrollPane();
+			cScroll.setName(CAPTION_NAME);
+			cScroll.setViewportView(getHexTableBody());
+			// Outer scroll size constraints
+			container.setMinimumSize(new Dimension(0, 280));
+			container.setPreferredSize(new Dimension(0, 280));
+			container.setMaximumSize(new Dimension(0, 280));
+			// Setup text field for ViewState info
+			vsInfoTxt = new JFormattedTextField();
+			vsInfoTxt.setEditable(false);
+			vsInfoTxt.setBackground(Color.decode("#D6D9DF"));
+			// Add to container
+			container.add(cScroll);
+			container.add(vsInfoTxt);
```

Comment:
Actually, better change the dimensions being set, the minimum and maximum could be removed and for the preferred size change the height to a lower value (100?).